client/src/api/v2/files.ts

```diff
@@ -24,25 +25,17 @@ export async function getFiles() {
 /**
  * Delete a log file
  *
- * @param {LogFile} file Log file
+ * @param file Log file
  */
-export async function deleteFile(file) {
+export async function deleteFile(file: LogFile) {
   await fetch(file.url, { method: 'DELETE' });
 }
 
-/**
- * @typedef {object} FilesHook
- * @property {?Array.<LogFile>}   files       List of log files
- * @property {function(LogFile)}  deleteFile  Delete a log file
- */
-
 /**
  * Use a list of log files
- *
- * @returns {FilesHook} Files hook
  */
 export function useFiles() {
```

Comment:
Not sure how to document the return type for this properly, as really there's two things returned. Should I make an interface for the return value? Feels kinda unnecessary.

Reply:
There's no need to annotate here. Typescript can infer the return type

Reply:
Oh not worried about return type, thinking more about what the returned values actually are i.e. "List of log files", "Delete a log file"

Reply:
Maybe something like "List of log files and delete old file function"?

Reply:
Or an interface is not a bad idea

Reply:
Happy to go with either. If you've got no preference I'll just describe it with a one-liner, but maybe for more complex return types we should consider an interface.

Reply:
I'm happy with that